packages/@aws-cdk/aws-route53/lib/cross-account-zone-delegation-handler/index.ts

```diff
@@ -22,13 +23,15 @@ export async function handler(event: AWSLambda.CloudFormationCustomResourceEvent
 }
 
 async function cfnEventHandler(props: ResourceProperties, isDeleteEvent: boolean) {
-  const { AssumeRoleArn, ParentZoneId, DelegatedZoneName, DelegatedZoneNameServers, TTL } = props;
+  const { AssumeRoleArn, ParentZoneId, ParentZoneName, DelegatedZoneName, DelegatedZoneNameServers, TTL } = props;
 
   const credentials = await getCrossAccountCredentials(AssumeRoleArn);
   const route53 = new Route53({ credentials });
 
+  const parentZoneId = ParentZoneId ?? await getHostedZoneIdByName(ParentZoneName!, route53);
```

Comment:
Seems we might want to error here if neither ZoneName or ZoneId are provided; I know we're guarded for that in the construct, but defense in depth...

Reply:
Sure sounds good. Will add in next change.